spec/Data_Parallelism.tex

```diff
@@ -405,6 +419,32 @@ \section{Promotion}
 is performed with getter methods~(\rsec{Getter_Methods}), field
 access can also be promoted.
 
+If the original function returns a value, the corresponding promoted
+expression is an iterator yielding each computed value in turn.
+
+When a promoted expression is used to initialize a variable, such as
+\chpl{var X = A.x;} in the above example, the variable is an array.
+The array's domain is defined by the expression that causes promotion:
+
+\begin{center}
+\begin{tabular}[c]{|l|l|}
+\hline
+input expression & resulting array's domain \\
+\hline
+array    &  that array's domain \\
+domain   &  that domain \\
+range    &  one-dimensional domain built from that range \\
+iterator &  1-based one-dimensional domain \\
+\hline
+\end{tabular}
+\end{center}
+
+\begin{future}
+If the iterator iterates over an array, a domain or a range,
+we may want the resulting array's domain to be the same as
+if the input expression were that array/domain/range.
```

Comment:
I think the real future work here is to allow iterator authors to specify the shape in their iterator. Say that here?